samples/nrf_desktop/src/hw_interface/motion_optical.c

```diff
@@ -752,27 +752,37 @@ static int init(struct config_options *options)
 	return err;
 }
 
-static void update_config(const u8_t config_id, const u8_t *data)
+static void update_config(const u8_t config_id, const u8_t *data, size_t size)
 {
-	switch (config_id) {
-	case CONFIG_EVENT_ID_MOTION_CPI:
-		atomic_set(&sensor_cpi, sys_get_le16(data));
+	if ((GROUP_FIELD_GET(config_id) != EVENT_GROUP_SETUP) ||
+	    (MOD_FIELD_GET(config_id) != SETUP_MODULE_SENSOR)) {
+		/* Not for us */
+		return;
+	}
+
+	switch (OPT_FIELD_GET(config_id)) {
+	case SENSOR_OPT_CPI:
+		__ASSERT_NO_MSG(size == sizeof(u32_t));
```

Comment:
Maybe add one assert above, since all parameters are u32 ? (Unless we want to add parameters with different size)

Reply:
This assert is wrong anyway... data is delivered from the outside so we must evaluate the size and handle error gracefully.

Reply:
Fixed